components/BottomAppBar/src/private/MDCBottomAppBarView.m

```diff
@@ -0,0 +1,250 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCBottomAppBarView.h"
+
+#import "MDCBottomAppBarAttributes.h"
+#import "MDCBottomAppBarLayer.h"
+#import "MaterialRTL.h"
+
+static NSString *kMDCBottomAppBarViewAnimKeyString = @"AnimKey";
+static NSString *kMDCBottomAppBarViewEmptyString = @"";
+static NSString *kMDCBottomAppBarViewPathString = @"path";
+static NSString *kMDCBottomAppBarViewPositionString = @"position";
+
+@interface MDCBottomAppBarCutView : UIView
+
+@end
+
+@implementation MDCBottomAppBarCutView
+
+// Allows touch events to pass through so MDCBottomAppBarController can handle touch events.
+- (UIView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event {
```

Comment:
Oy. We have so many custom classes that do this. I wish we could just be like `.touchPassesThru = YES` on any UIView.

Reply:
Yes, we'll have to figure out how to address this once the bottom drawer is created. We can remove this if another way to allow touch events to pass through is created.